app/views/oaisys/pmh/list_identifiers.xml.builder

```diff
@@ -0,0 +1,33 @@
+xml.push_deferred_attribute('xmlns:dc': 'http://purl.org/dc/elements/1.1/',
+                            'xmlns:oai-id': 'http://www.openarchives.org/OAI/2.0/oai-identifier',
+                            'xmlns': 'http://www.openarchives.org/OAI/2.0/',
+                            'xmlns:etd_ms': 'http://www.ndltd.org/standards/metadata/etdms/1.0/',
+                            'xmlns:atom': 'http://www.w3.org/2005/Atom',
+                            'xmlns:rdfs': 'http://www.w3.org/2000/01/rdf-schema#',
+                            'xmlns:rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+                            'xmlns:dcterms': 'http://purl.org/dc/terms/',
+                            'xmlns:oreatom': 'http://www.openarchives.org/ore/atom/',
+                            'xmlns:oai_dc': 'http://www.openarchives.org/OAI/2.0/oai_dc/',
+                            'xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+                            'xsi:schemaLocation': 'http://www.openarchives.org/OAI/2.0/ '\
+                            'http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd '\
+                            'http://www.openarchives.org/OAI/2.0/oai-identifier '\
+                            'http://www.openarchives.org/OAI/2.0/oai-identifier.xsd '\
+                            'http://www.openarchives.org/OAI/2.0/oai_dc/ '\
+                            'http://www.openarchives.org/OAI/2.0/oai_dc.xsd '\
+                            'http://www.ndltd.org/standards/metadata/etdms/1.0/ '\
+                            'http://www.ndltd.org/standards/metadata/etdms/1-0/etdms.xsd '\
+                            'http://www.w3.org/2005/Atom')
+
+xml.tag!('request', parameters, 'https://era.library.ualberta.ca/oai')
+xml.ListIdentifiers do
+  identifiers.each do |identifier|
```

Comment:
Ok, neat trick you can use so that you don't have to index into the array by position (which then forces people go and look in pmh_controller to see what data is in which position)

Ruby will automatically destructure an array to populate assignments, so if (like here) you have an array of arrays `array_of_arrays = [[1, 'abc', :some_other_data], [2, 'xyz', :another_symbol]]`

Instead of:

```ruby
array_of_arrays.each do |array|
   id = array[0]
   name = array[1]
   payload = array[2]
   # use the data ....
end
```

You can instead directly destructure the inner arrays:

```ruby
array_of_arrays.each do |id, name, payload|
   # use the data ....
end
```

it pulls apart the inner array and populates the named variables by position, meaning people don't have to look elsewhere to see what data is what.

I learned this from Nick Quantro's excellent [Stupid Ruby Tricks](https://thoughtbot.com/blog/stupid-ruby-tricks)

Reply:
Interesting, that's a neat trick for sure! Made the change.